compute_tools/src/compute.rs

```diff
@@ -343,4 +345,71 @@ impl ComputeNode {
         self.prepare_pgdata()?;
         self.run()
     }
+
+    // Look for core dumps and collect backtraces.
+    //
+    // EKS worker nodes have following core dump settings:
+    //   /proc/sys/kernel/core_pattern -> core
+    //   /proc/sys/kernel/core_uses_pid -> 1
+    //   ulimint -c -> unlimited
+    // which results in core dumps being written to postgres data directory as core.<pid>.
+    //
+    // For local testing I also added support for lldb, which writes core dumps to /cores.
+    fn check_for_core_dumps(&self) {
+        let core_dump_dir = if std::env::consts::OS == "macos" {
+            std::path::Path::new("/cores/")
+        } else {
+            std::path::Path::new(&self.pgdata)
+        };
+
+        info!(
+            "checking for core dumps in {}",
+            core_dump_dir.to_string_lossy()
+        );
+
+        let paths = fs::read_dir(core_dump_dir).expect("cannot list pgdata for core dumps");
+        for path in paths {
+            if let Some(name) = path
+                .expect("can't read pgdata directory entry")
+                .path()
+                .file_name()
+            {
+                let name = name.to_string_lossy();
+                if name.starts_with("core.") {
+                    warn!("core dump found: {}, collecting backtrace", name);
+                    let full_core_path = std::path::Path::new(core_dump_dir).join(name.as_ref());
+                    let full_core_path = full_core_path.to_string_lossy();
+
+                    let backtrace = if std::env::consts::OS == "macos" {
+                        Command::new("lldb")
```

Comment:
Another option would be to try using both, one after another. There's gdb for mac and lldb for linux.

Reply:
Was thinking of that too, but didn't come up with a nice way to test if debugger is available or not. There is https://crates.io/crates/which but that is a bit too heavyweight.

Reply:
Or you could ignore error kind completely and just first try lldb and then gdb and then bail with error.

Reply:
> Oh, you don't have to! We could just catch execve's error:

Nice, rewrote it this way